File: 169-majority-element/169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        
        count = 0
        majorCandidate = None
        
        for num in nums:
            if count == 0:
                majorCandidate = num
            
            count += 1 if num == majorCandidate else -1
        
        return majorCandidate

File: 169-majority-element/169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        
        int count = 0;
        int cand = 0;
        
        for (int num:nums){
            if (count == 0){
                cand = num;
            }
            if(num==cand){
                count++;
            }
            else{
                count--;
            }
        }
        return cand;
        
    }
}